internal/knowledge/graph_binder.go

```diff
@@ -15,13 +15,21 @@ func NewGraphBinder(graph *Graph) *GraphBinder {
 }
 
 // Relate relate one asset to another
-func (gb *GraphBinder) Relate(from string, relationType schema.RelationType, to string) {
-	fromAsset := gb.graph.AddAsset(relationType.FromType, from)
-	toAsset := gb.graph.AddAsset(relationType.ToType, to)
+func (gb *GraphBinder) Relate(from string, relationType schema.RelationType, to string) error {
+	fromAsset, err := gb.graph.AddAsset(relationType.FromType, from)
+	if err != nil {
+		return err
```

Comment:
I suggest wrapping the error so that it's more explicit where the error comes from when it happens.

Reply:
Done